Tools/visualize_log.py

```diff
@@ -1,221 +1,211 @@
 
-    # Read the event CSV file.
-    event_df = pd.read_csv(event_file_path)
+def log_event_summary(event_df: pd.DataFrame) -> None:
+    """Logs a summary of the events.
 
-    # Sanitize the "Event" column to ensure consistency.
-    event_df["Event"] = event_df["Event"].str.upper().str.strip()
+    Args:
+        event_df: Dataframe containing the events.
+    """
+    # Log total number of events.
+    total_events = len(event_df)
+    logging.info("Total number of events: %d.", total_events)
 
-    # Debugging: Print unique event types to verify correct parsing.
-    unique_events = event_df["Event"].unique()
-    print(f"Unique Events Found: {unique_events}")
+    # Log the counts of each event type.
+    event_counts = event_df["Event"].value_counts()
+    logging.info("Event counts:")
+    for event_type, count in event_counts.items():
+        logging.info("  %s: %d", event_type, count)
 
-    # Create a 3D plot.
-    fig = plt.figure(figsize=(14, 10))
-    ax = fig.add_subplot(111, projection="3d")
+    # Calculate the time duration of the events.
+    start_time = event_df["Time"].min()
+    end_time = event_df["Time"].max()
+    duration = end_time - start_time
+    logging.info("Total duration of events: %.2f seconds (from %.2f to %.2f).",
+                 duration, start_time, end_time)
+
+    # Determine the times of the hits and misses.
+    hits = event_df[event_df["Event"] == "INTERCEPTOR_HIT"]
+    misses = event_df[event_df["Event"] == "INTERCEPTOR_MISS"]
+    logging.info("Number of hits recorded: %d.", len(hits))
+    if not hits.empty:
+        first_hit_time = hits["Time"].min()
+        last_hit_time = hits["Time"].max()
+        logging.info("  First hit at %.2f, last hit at %.2f.", first_hit_time,
+                     last_hit_time)
+    logging.info("Number of misses recorded: %d.", len(misses))
+    if not misses.empty:
+        first_miss_time = misses["Time"].min()
+        last_miss_time = misses["Time"].max()
+        logging.info("  First miss at %.2f, last miss at %.2f.",
+                     first_miss_time, last_miss_time)
+
+
+def plot_telemetry(telemetry_df: str, event_df: str) -> None:
+    """Plots the trajectories in the telemetry data and the events.
+
+    Args:
+        telemetry_df: Dataframe containing the telemetry data.
+        event_df: Dataframe containing the events.
+    """
+    fig, ax = plt.subplots(figsize=(16, 8), subplot_kw={"projection": "3d"})
 
     # Define colors for different agent types.
-    colors = {"T": "red", "M": "blue"}
+    colors = {
+        "M": "blue",
+        "T": "red",
+    }
 
-    # Group data by AgentID.
-    agent_types = set()
-    for agent_id, agent_data in df.groupby("AgentID"):
+    # Plot the agent trajectories.
+    for _, agent_data in telemetry_df.groupby("AgentID"):
         agent_type = agent_data["AgentType"].iloc[0]
         color = colors.get(agent_type, "black")
-        downsampled = agent_data.iloc[::10]
```

Comment:
Used to be downsampled cause there would be way too much data and the 3d viz would be laggy. Double checking that its intentional to remove data downsample.

Reply:
Yes, I was able to successfully plot with 2_ddg_5_swarms_100_ucav.pbtxt without any issues.  Also the simulator only logs telemetry every 0.1 seconds.